### DISCUSSION TITLE
The plural of SCHEMA is SCHEMATA. Schemas sounds stupid.

### DISCUSSION THREAD

**Discussion by allosaurus123** [Automattic/mongoose#11218] [Q&A] [answered]
https://mongoosejs.com/docs/guide.html
^ The title at the top of the page is "Schemas," but this is the less academic and, frankly, vulgar plural form of schema. The prescriptive plural of schema is **schemata**.

**Comment by allosaurus123** [comment]
I realize programmers are often bad writers, but there's no need to degrade English.

**Comment by Uzlopak** [comment]
According to the Oxford Dictionary schemas is a valid plural.

https://www.oxfordlearnersdictionaries.com/definition/english/schema?q=schema

**Reply by vkarpov15** [reply]
If 'schemas' can be considered valid, I'd prefer to keep it. There's a lot of Mongoose devs that aren't native English speakers, so using a more academic plural form as opposed to the more common "just add an 's' at the end" form would be counterproductive.

**Reply by allosaurus123** [reply]
https://www.merriam-webster.com/dictionary/ain%27t
^ **_ain't_** is also in the dictionary. That doesn't make it standard.

**Reply by allosaurus123** [reply]
Also, the user above, Uzlopak, cited an English _learner's_ dictionary. Hardly an ideal reference in a professional environment!

**Reply by Uzlopak** [reply]
Btw. I brought a proof saying schemas is valid plural. You argue that this is not a valid source and schemas a "vulgar" plural of schema. So please bring a proof for the claim you make.

**Reply by allosaurus123** [reply]
Well, first of all, you cited an English _learner's_ dictionary. Secondly, I noticed that you are not a native speaker of English, so I wouldn't expect you to be conscious of these little subtleties, anyway. (Would it be professional to refer to you as _Alter_ in German? _Wahrscheinlich nicht_. The same guidelines apply to English.) I'm unsure of your background, but I assume you are not a linguist. I am. "Schema" is a Greek loanword, and it carries Greek plurality. In more reputable dictionaries and lexica, you will notice that "schemata" precedes the more vulgar "schemas."^1,2,3 The same pattern is followed in stigma --> stigmata, stoma --> stomata, lemma --> lemmata, et al.

1. The Cambridge Dictionary: https://dictionary.cambridge.org/us/dictionary/english/schema
2. Wiktionary: https://dictionary.cambridge.org/us/dictionary/english/schema
3. Britannica: https://dictionary.cambridge.org/us/dictionary/english/schema](https://www.britannica.com/science/schema-cognitive)

The point, of course, is that "schemas," while acceptable because of lay usage, is neither professional nor ideal in a professional setting. In the same way the educated wince when a person uses "that is so cringe" instead of the more prescriptive "that is so cringeworthy," the cognoscente will notice the lazy use of "schemas." I did. Others will, too.

**Reply by allosaurus123** [reply]
In reference to your deleted comment, personally, I think a professional would be willing to admit when he is wrong. But what do I know? I'm just a linguist/engineer. ;-)

**Reply by Uzlopak** [reply]
Please fix your links. All of them are linking to the Cambridge dictionary. Also there was atleast in the Cambridge Dictionary no proof that schemas would be the "vulgar" variant. If we go academic then we need primary sources.

**Comment by allosaurus123** [comment]
Uzlopak,

I have corrected the links. Here:

2. Wiktionary: https://en.wiktionary.org/wiki/schema
3. Britannica: https://www.britannica.com/science/schema-cognitive

I went ahead and added some more:

4. Lexico (powered by Oxford): https://www.lexico.com/en/definition/schema
5. Merriam-Webster: https://www.merriam-webster.com/dictionary/schema

And here is a German-English dictionary, which is particularly useful to you, as a germanophone:
6. https://dict.leo.org/german-english/schema

In each of these reputable dictionaries, you will notice that "schemata" takes precedence. I find it strange that you've elected to argue with me, considering the plural of _das Schema_ in German is also _die Schemata_. Like German, English has many irregular plurals, such as datum --> data.

There is an analytic trend in English to simplify vocabulary and grammar. It's why "thou" (and its derivatives "thee", "thy", and "thine") have fallen out of common usage, and why many anglophones are unable to explain when to use "who" vs. "whom" properly. This is a natural change in English. It would be out-dated, however, to use the older spelling of "connexion" instead of the now-prescriptive and newer "connection," since it has been at least a century since the older spelling was even used in professional literature.

But this is not the case for "schemata," which is still in common usage in academic and professional settings. Since computer science derives most of its vocabulary from academia, it is therefore important to use the most prescriptive register--especially in the documentation of something like Mongoose.

You can read more about prestige variants and prescriptive language in _American English_, Second Edition, by Wolfram and Schilling-Estes.

If you're the type of person who insists on saying, "data is," instead of the correct, "data are," then you wouldn't understand it, anyway, and I'm wasting my time. The plural form--the prescriptive and original plural--is "schemata."

**Reply by Uzlopak** [reply]
Please provide a PR.

**Reply by Uzlopak** [reply]
Btw. I dont see any return of investment in renaming something into a more "prestigious" variant. So I really encourage you to provide a PR. If it makes you sleep better, why not.

**Reply by Uzlopak** [reply]
I read the comment you deleted. Tbh I think this is not the spirit of OSS. If you are just about nagging some wording and want that others do the work than you will not achieve anything. Tbh I wonder why you even registered yourself to GitHub. Was it only to nag about schemas vs. schemata in the mongoose documentation?

Btw. I had already worse experience regarding spelling. E.g. organisation instead of organization in a project where everything else was written in American English. That was really annoying.

**Comment by Uzlopak** [comment, marked as answer]
Won't fix
